### PR TITLE
Reworking ServerTaps Core Systems (and adding a Reload Command)

### PR DESCRIPTION
**Pull request by Velyn-N**
# This depends on #217!!!

@srmullaney and I spend some time reworking the Core Systems of ServerTap. That includes:

- Moving Files around
- Renaming things
- Providing Loggers & Config through Constructors (good for testing purposes if ever needed)
- Replacing public Fields with Getters
- Removing static Method calls
- Refactoring WebHooks Logic
- Refactoring WebServer (Javalin) Setup
- Refactoring Route Configuration
- Refactoring URL-Blocking (originally from #214, removed Regex because of Performance and Regex bad)
- Refactored Main (and gave it a ~~better~~new name)

We also made a Command (`/servertap`) that provides reload and info for ServerTap.

We did not touch the API V1 Implementations whenever possible. We might want to in the Future to further reduce getInstance() calls. They make Code hard to follow.

### CODE REVIEW

**Comment by phybros** [comment]
Fantastic!!

When you said 

> We did not touch the API V1 Implementations whenever possible

Does that mean you did make some v1 changes?

**Comment by srmullaney** [comment]
Yes! We worked on refactoring the cord ServerTap code, cleaned things up, etc. We did not make changes to the V1 API in this PR. We are planning on doing a V2 API where we refactor everything to do with the API.

**Comment by phybros** [comment]
OK, I tested a bunch of different stuff. Webhooks, websockets, TLS, some basic stuff like swagger etc.

As far as I can tell everything is working great! Going to do a little bit more testing and then I'm good to merge

**Comment by srmullaney** [comment]
Awesome! I believe #217 needs to be merged first as we did everything on the same branch.

**Comment by phybros** [comment]
Nah I think this branch/PR includes all those changes as well if you started from there. For example https://github.com/phybros/servertap/pull/236/files#diff-8a9174254c8614cc18da0da58a8e3f176f33bba5b5a062b764f8d26ef9b029a6

**Comment by srmullaney** [comment]
Ah okay. I was just going off what he had said. I think you are right but I am not a git expert and don't want to mess it up.
